internal/backend/twitter.go

```diff
@@ -0,0 +1,184 @@
+	return twitterUsers, err
+}
+
+func (t *twitterClient) LookupUsers(screenNames []string) ([]twitter.User, error) {
+	twitterUsers, _, err := t.client.Users.Lookup(&twitter.UserLookupParams{
+		ScreenName: screenNames,
+	})
+	return twitterUsers, err
+}
+
+func (d *delphisBackend) GetTwitterAccessToken(ctx context.Context) (string, string, error) {
+	/* Get the authed user */
+	authedUser := auth.GetAuthedUser(ctx)
+	if authedUser == nil {
+		return "", "", fmt.Errorf("Need auth")
+	}
+
+	/* Obtain authed user profile */
+	authedUserProfile, err := d.db.GetUserProfileByUserID(ctx, authedUser.UserID)
+	if err != nil {
+		return "", "", err
+	}
+
+	/* Obtain authed user social info  */
+	authedSocialInfo, err := d.db.GetSocialInfosByUserProfileID(ctx, *&authedUserProfile.ID)
+	if err != nil {
+		return "", "", err
+	}
+
+	/* Extract tokens from social info */
+	accessToken := ""
+	accessTokenSecret := ""
+	for _, info := range authedSocialInfo {
+		if strings.ToLower(info.Network) == "twitter" {
+			accessToken = info.AccessToken
+			accessTokenSecret = info.AccessTokenSecret
+		}
+	}
+
+	return accessToken, accessTokenSecret, nil
+}
+
+func (d *delphisBackend) GetTwitterClient(ctx context.Context) (TwitterClient, error) {
+	/* Obtain infos needed for creating Twitter API client */
+	consumerKey := d.config.Twitter.ConsumerKey
+	consumerSecret := d.config.Twitter.ConsumerSecret
+	accessToken, accessTokenSecret, err := d.GetTwitterAccessToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	/* Check that everything is ready to go */
+	if len(consumerKey) == 0 || len(consumerSecret) == 0 || len(accessToken) == 0 || len(accessTokenSecret) == 0 {
+		return nil, fmt.Errorf("There is a problem retrieving authed user Twitter data")
+	}
+
+	/* Create client object */
+	config := oauth1.NewConfig(consumerKey, consumerSecret)
+	token := oauth1.NewToken(accessToken, accessTokenSecret)
+	httpClient := config.Client(oauth1.NoContext, token)
+	client := twitter.NewClient(httpClient)
+	return &twitterClient{
+		client: client,
+	}, nil
+}
+
+func (d *delphisBackend) GetTwitterUserHandleAutocompletes(ctx context.Context, twitterClient TwitterClient, attempt string) ([]string, error) {
+	/* Fetch autocompletes result eagerly from twitter APIs. A connection-based paging
+	   system would have more quality but would also introduce additional overhead.
+	   As a tradeoff we limit the number of pages fetched by assuming that the best
+	   results will be on top of the list */
+	var results []string
+	curPage := 0
+	for resultSize := 0; (curPage == 0 || resultSize == twitterAutocompletesPageSize) && curPage < twitterAutocompletsMaxPages; curPage++ {
+		twitterUsers, err := twitterClient.SearchUsers(attempt, curPage, twitterAutocompletesPageSize)
+		if err != nil {
+			return nil, err
+		}
+		resultSize = len(twitterUsers)
+		for _, twitterUser := range twitterUsers {
+			results = append(results, twitterUser.ScreenName)
```

Comment:
Presumably we can just update this function as we have access to the full user here.